pkg/registry/common/nslabelsmatch/client_test.go

```diff
@@ -0,0 +1,114 @@
+// Copyright (c) 2021 Doc.ai and/or its affiliates.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package nslabelsmatch_test
+
+import (
+	"context"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"go.uber.org/goleak"
+
+	"github.com/networkservicemesh/api/pkg/api/registry"
+
+	"github.com/networkservicemesh/sdk/pkg/registry/common/memory"
+	"github.com/networkservicemesh/sdk/pkg/registry/common/nslabelsmatch"
+	"github.com/networkservicemesh/sdk/pkg/registry/core/adapters"
+	"github.com/networkservicemesh/sdk/pkg/registry/core/chain"
+)
+
+const (
+	testEnvName                      = "NODE_NAME"
+	testEnvValue                     = "testValue"
+	destinationTestKey               = "nodeName"
+	destinationTestCorrectTemplate   = `{{index .Src "NodeNameKey"}}`
+	destinationTestIncorrectTemplate = `{{index .Src "SomeRandomKey"}}`
+	destinationTestNotATemplate      = "SomeRandomString"
+)
+
+func TestNSTemplates(t *testing.T) {
+	err := os.Setenv(testEnvName, testEnvValue)
+	require.NoError(t, err)
+
+	tests := []struct {
+		name     string
+		template string
+		want     string
+	}{
+		{
+			name:     "CorrectTemplate",
+			template: destinationTestCorrectTemplate,
+			want:     testEnvValue,
+		},
+		{
+			name:     "IncorrectTemplate",
+			template: destinationTestIncorrectTemplate,
+			want:     "",
+		},
+		{
+			name:     "NotATemplate",
+			template: destinationTestNotATemplate,
+			want:     destinationTestNotATemplate,
+		},
+	}
+
+	for _, tc := range tests {
+		tc := tc
```

Comment:
This assigning disables GoLand users from running separate table tests just by clicking on them :)

I know that linter starts screaming if you are trying to make something like this:
```go
for _, obj := rang objs {
    doSomething(func() {
        doSomethingElse(obj.a, obj.b)
    })
}
```
Because if `doSomething` doesn't block, all runs of `doSomethingElse` will receive the last `obj` from the `objs`.

But it is totally OK to do such thing with `t.Run`, because it is waiting for the passed function result. So please just add here `// nolint:scopelint` instead of `tc := tc`.